### PR TITLE
IPDK: Optimize docker image based on user option

### PR DESCRIPTION
**Pull request by n-sandeep**
As part of docker build, after successfully starting p4ovs via
start_p4ovs.sh option is given to user for reducing this memory by setting
'KEEP_SOURCE_CODE' variable in docker.env file.
a) If 'KEEP_SOURCE_CODE=NO' then:
   i) Copy few dependent scripts from P4-OVS repository to scripts directory.
   ii) Removing complete source code for P4-OVS repository.
   iii) Removing complete source code of p4-driver repository.
   iv) Removing complete source code of P4C.
   v) Removing source code of dependent packages installed as part of
      install_dep_packages.sh script.
b) If 'KEEP_SOURCE_CODE=YES' then:
   i) Removing object files via make clean for P4-OVS repository.
   ii) Removing object files via make clean for p4-driver repository (**TODO).
   iii) Removing object files via make clean for P4C.
   iv) Removing source code of dependent packages installed as part of
       install_dep_packages.sh script.
c) If none of the above, then keep the source code as is.

Default value is 'KEEP_SOURCE_CODE=NO'.

Change-type: ImplementationChange
Title: IPDK: Optimize docker image based on user option
Signed-off-by: n-sandeep <sandeep.nagapattinam@intel.com>

### CODE REVIEW

**Comment by mestery** [comment]
I wonder if it makes sense to look at pushing out the base image (which takes a long time to build and is huge) to a docker repository somewhere (such as DockerHub).

**Comment by mestery** [comment]
Can you review #31 so we can merge that, and then you can pull my changes in there and add the equivalent changes into the Ubuntu infrastructure Makefile/Dockerfile as well?

File: build/IPDK_Container/Dockerfile
Comment (mestery) [inline build/IPDK_Container/Dockerfile:70]: Can you make similar changes to the Ubuntu20.04/Dockerfile and Makefile as well? I merged some changes today in the Dockerfile, so make sure to pick those up first.

**Review by mestery — CHANGES_REQUESTED** [review]
Looks fine, but can you update the Ubuntu20.04/[Dockerfile,Makefile] to match before merging?

File: build/IPDK_Container/Dockerfile
Comment (vsureshkumarp) [reply build/IPDK_Container/Dockerfile:70]: @mestery, Planning to pick up the Ubuntu container immediately once after the review process and considerable testing is done for Fedora container.

File: build/IPDK_Container/README
Comment (vsureshkumarp) [inline build/IPDK_Container/README:39]: Need to modify the README. It is better to add the importance of  KEEP_SOURCE_CODE and its details in docker.env.  README just needs to provide how to use and its supported values of the KEEP_SOURCE_CODE ENV variable.

File: build/IPDK_Container/README
Comment (vsureshkumarp) [inline build/IPDK_Container/README:117]: We need to further optimize this procedure later to make it common instead of following different paths of the scripts.

**Review by mestery — APPROVED** [review]
LGTM, look forward to the Ubuntu changes once this one merges!

File: build/IPDK_Container/scripts/run_cleanup.sh
Comment (stolsma) [inline build/IPDK_Container/scripts/run_cleanup.sh:14]: Why not always copy the scripts to the scripts dir? In that way there is no need for different paths for starting and using just like @vsureshkumarp mentioned above.

File: build/IPDK_Container/Dockerfile
Comment (n-sandeep) [reply build/IPDK_Container/Dockerfile:70]: Done, made changes on top of #31 for Ubuntu and validated with both KEEP_SOURCE_CODE=YES  and KEEP_SOURCE_CODE=NO options.

File: build/IPDK_Container/README
Comment (n-sandeep) [reply build/IPDK_Container/README:39]: Done updated README as suggested.

**Comment by stolsma** [comment]
@n-sandeep Nice changes to one workflow! Although I have nothing to say about it: lgtm! 😄

**Review by vsureshkumarp — APPROVED** [review]

**Pull request merged**
